# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmOpIter.h
# Commit: 546d74acfb81
# Full Hash: 546d74acfb81497dc00809fd4e4096edfe857ef5
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2020-08-19 10:01:16
# Regressor Bug: 1561521
# File Overlap Count: 2
# Description:
#   Bug 1561521 - Add non-nullable references. r=lth
#   
#   This commit adds decoding, validation, and JS-API type-checking for
#   non-nullable references.
#   
# ==============================================================================

diff -r e055945526a2 -r 546d74acfb81 js/src/wasm/WasmOpIter.h
--- a/js/src/wasm/WasmOpIter.h	Tue Aug 18 16:58:55 2020 +0000
+++ b/js/src/wasm/WasmOpIter.h	Tue Aug 18 16:58:59 2020 +0000
@@ -2309,7 +2309,7 @@
     }
   }
 
-  return push(RefType::fromTypeIndex(*typeIndex));
+  return push(RefType::fromTypeIndex(*typeIndex, false));
 }
 
 template <typename Policy>
@@ -2328,7 +2328,7 @@
     return false;
   }
 
-  if (!popWithType(RefType::fromTypeIndex(*typeIndex), ptr)) {
+  if (!popWithType(RefType::fromTypeIndex(*typeIndex, true), ptr)) {
     return false;
   }
 
@@ -2360,7 +2360,7 @@
     return fail("field is not mutable");
   }
 
-  if (!popWithType(RefType::fromTypeIndex(*typeIndex), ptr)) {
+  if (!popWithType(RefType::fromTypeIndex(*typeIndex, true), ptr)) {
     return false;
   }
 